php/queryads.php

```diff
@@ -37,7 +34,16 @@ function is_valid_domain_name($domain_name)
     die();
 }
 
-$proc = popen("sudo pihole -q ".$url, 'r');
+if(isset($_GET["exact"]))
+{
+    $exact = "-exact";
+}
+else
+{
+    $exact = "";
+}
```

Comment:
Good job preventing a security problem! 👍 If we just passed in the value of `exact`, that wouldn't be good.